core/src/main/java/de/danoeh/antennapod/core/feed/util/ImageResourceUtils.java

```diff
@@ -16,46 +20,72 @@ private ImageResourceUtils() {
     /**
      * returns the image location, does prefer the episode cover if available.
      */
-    public static String getEpisodeImageLocation(ImageResource resource) {
-        return resource.getImageLocation();
+    @Nullable
+    public static String getEpisodeImageLocation(@NonNull Playable playable) {
+        return playable.getImageLocation();
     }
 
+    /**
+     * returns the image location, does prefer the episode cover if available.
+     */
+    @Nullable
+    public static String getEpisodeImageLocation(@NonNull FeedItem feedItem) {
+        return feedItem.getImageLocation();
+    }
 
     /**
      * returns the image location, does prefer the episode cover if available and enabled in settings.
      */
-    public static String getEpisodeListImageLocation(ImageResource resource) {
-
+    @Nullable
+    public static String getEpisodeListImageLocation(@NonNull Playable playable) {
         if (UserPreferences.getUseEpisodeCoverSetting()) {
-            return resource.getImageLocation();
+            return playable.getImageLocation();
         } else {
-            return getShowImageLocation(resource);
+            return getFallbackImageLocation(playable);
         }
     }
 
-    public static String getFallbackImageLocation(ImageResource resource) {
-        return getShowImageLocation(resource);
+    /**
+     * returns the image location, does prefer the episode cover if available and enabled in settings.
+     */
+    @Nullable
+    public static String getEpisodeListImageLocation(@NonNull FeedItem feedItem) {
+        if (UserPreferences.getUseEpisodeCoverSetting()) {
+            return feedItem.getImageLocation();
+        } else {
+            return getFallbackImageLocation(feedItem);
+        }
     }
 
-    private static String getShowImageLocation(ImageResource resource) {
+    /**
+     * returns the image location, does prefer the episode cover if available and enabled in settings.
+     */
+    @Nullable
+    public static String getEpisodeListImageLocation(@NonNull Feed feed) {
```

Comment:
Can't callers just use the method directly? There is no setting involved here.

Reply:
I removed that trivial method now. Shall I remove the two trivial `getEpisodeImageLocation()` methods, too?

Reply:
How do you feel about it? I personally think that it could be good to remove them. We have a ton of `Utils` classes already and it's pretty hard to keep track of them. If we can reduce the number of usages, the code probably becomes more readable.

Reply:
Great, I removed the trivial methods, too. I like to have a smaller code base with the same functionality.